lib/dgram.js

```diff
@@ -344,6 +346,13 @@ Socket.prototype.send = function(buffer,
   if (typeof callback !== 'function')
     callback = undefined;
 
+  if (typeof address === 'function') {
+    callback = address;
+    address = undefined;
+  } else if (typeof address !== 'undefined' && typeof address !== 'string') {
+    throw new TypeError('"Address" argument must be a string or undefined');
```

Comment:
I'll take a closer look at the wording and punctuation.  IMO we should be shooting for consistency with core error messaging; not inconsistent error messaging which may be used elsewhere in this module.